src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -2,19 +2,109 @@ package cromwell.binding
+  override val allWorkflows = importedWorkflows.toSet
+}
+
+/**
+ * Represents a WdlNamespace which has a local workflow, i.e. a directly runnable namespace
+ *
+ * FIXME: getCallFromMemberAccessAst was being doubly used as validator *and* runtime fetcher, thus needs that formatter. We shouldn't be doing syntax checks at runtime
+ */
+case class NamespaceWithWorkflow(importedAs: Option[String],
+                                 workflow: Workflow,
+                                 imports: Seq[Import],
+                                 namespaces: Seq[WdlNamespace],
+                                 tasks: Seq[Task],
+                                 importedWorkflows: Seq[Workflow],
+                                 terminalMap: Map[Terminal, WdlSource],
+                                 wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter,
+                                 ast: Ast) extends WdlNamespace {
+  override val allWorkflows = Set(workflow) ++ importedWorkflows
+
+  /**
+   * Confirm all required inputs are present and attempt to coerce raw inputs to `WdlValue`s.
+   * This can fail if required raw inputs are missing or if the values for a specified raw input
+   * cannot be coerced to the target type of the input as specified in the namespace.
+   */
+  def coerceRawInputs(rawInputs: WorkflowRawInputs): Try[WorkflowCoercedInputs] = {
+    def coerceRawInput(fqn: FullyQualifiedName, wdlType: WdlType): Try[WdlValue] = fqn match {
+      case _ if rawInputs.contains(fqn) =>
+        val rawInput = rawInputs.get(fqn).get
+        wdlType.coerceRawValue(rawInput).recoverWith {
+          case e => Failure(new UnsatisfiedInputsException(s"Failed to coerce input $fqn value $rawInput of ${rawInput.getClass} to $wdlType."))
+        }
+      case _ => Failure(new UnsatisfiedInputsException(s"Required workflow input '$fqn' not specified."))
+    }
+
+    val tryCoercedValues = workflow.inputs map {case (fqn, wdlType) => fqn -> coerceRawInput(fqn, wdlType)}
+
+    val (successes, failures) = tryCoercedValues.partition { case (_, tryValue) => tryValue.isSuccess }
+    if (failures.isEmpty) {
+      Try(successes.map { case (key, tryValue) => key -> tryValue.get })
+    } else {
+      val message = failures.values.collect { case f: Failure[_] => f.exception.getMessage }.mkString("\n")
+      Failure(new UnsatisfiedInputsException(message))
+    }
+  }
+
+  /*
+    FIXME: Originally this was called 2x - in validation and in WorkflowManagerActor. In the first case, the scaladoc
+    comment makes sense. In the latter case it doesn't seem to be a check. Is the validating circumstance checking via
+    side effect? Or the other way around?
+
+    TODO/FIXME: Is this really the right thing to be running here anyways? WTF should syntax be checked at runtime? We could get rid of the formatter
+   */
+  /** Partially evaluate MemberAccess ASTs to make sure they're not nonsense at compile time */
```

Comment:
Regarding the syntax error formatter... it's only really needed so we can craft error messages which use show the problematic and relevant lines of source code, so the syntax error formatter needs access to the WDL source code at the very least to make this possible.  This kind of has the side effect of needing to be passed around a bunch... I don't know a better way to do it off the top of my head